internal/elasticsearch/enrich/schema.go

```diff
@@ -0,0 +1,152 @@
+			"id": resourceschema.StringAttribute{
+				MarkdownDescription: "Internal identifier of the resource",
+				Computed:            true,
+			},
+			"name": resourceschema.StringAttribute{
+				MarkdownDescription: "Name of the enrich policy to manage.",
+				Required:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					stringvalidator.LengthBetween(1, 255),
+				},
+			},
+			"policy_type": resourceschema.StringAttribute{
+				MarkdownDescription: "The type of enrich policy, can be one of geo_match, match, range.",
+				Required:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					stringvalidator.OneOf("geo_match", "match", "range"),
+				},
+			},
+			"indices": resourceschema.ListAttribute{
+				MarkdownDescription: "Array of one or more source indices used to create the enrich index.",
+				ElementType:         types.StringType,
+				Required:            true,
+				PlanModifiers: []planmodifier.List{
+					listplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.List{
+					listvalidator.SizeAtLeast(1),
+				},
+			},
+			"match_field": resourceschema.StringAttribute{
+				MarkdownDescription: "Field in source indices used to match incoming documents.",
+				Required:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					stringvalidator.LengthBetween(1, 255),
+				},
+			},
+			"enrich_fields": resourceschema.ListAttribute{
+				MarkdownDescription: "Fields to add to matching incoming documents. These fields must be present in the source indices.",
+				ElementType:         types.StringType,
+				Required:            true,
+				PlanModifiers: []planmodifier.List{
+					listplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.List{
+					listvalidator.SizeAtLeast(1),
+				},
+			},
+			"query": resourceschema.StringAttribute{
+				MarkdownDescription: "Query used to filter documents in the enrich index. The policy only uses documents matching this query to enrich incoming documents. Defaults to a match_all query.",
+				Optional:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					stringvalidator.RegexMatches(regexp.MustCompile(`^\{.*\}$`), "must be valid JSON"),
+				},
+			},
+			"execute": resourceschema.BoolAttribute{
+				MarkdownDescription: "Whether to call the execute API function in order to create the enrich index.",
+				Optional:            true,
+				Computed:            true,
+				Default:             booldefault.StaticBool(true),
+				PlanModifiers: []planmodifier.Bool{
+					boolplanmodifier.RequiresReplace(),
+				},
+			},
+		},
+	}
+}
+
+func GetDataSourceSchema() schema.Schema {
```

Comment:
Similarly, let's move this to `data_source.go` and move the resource schema to `resource.go`